bookmarks/src/commonMain/kotlin/catchup/bookmarks/BookmarkRepository.kt

```diff
@@ -0,0 +1,113 @@
+package catchup.bookmarks
+
+import app.cash.sqldelight.Query
+import app.cash.sqldelight.Transacter
+import app.cash.sqldelight.coroutines.asFlow
+import app.cash.sqldelight.db.QueryResult
+import app.cash.sqldelight.db.SqlCursor
+import catchup.bookmarks.db.Bookmark
+import catchup.bookmarks.db.CatchUpDatabase
+import catchup.service.api.CatchUpItem
+import catchup.service.api.toCatchUpItem
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock.System
+import kotlinx.datetime.Instant
+
+interface BookmarkRepository {
+  fun addBookmark(id: Long, timestamp: Instant = System.now())
+
+  fun removeBookmark(id: Long)
+
+  fun isBookmarked(id: Long): Flow<Boolean>
+
+  fun bookmarksCountFlow(): Flow<Long>
+
+  // Exposed to create a PagingSource
+  fun bookmarksCountQuery(): Query<Long>
+
+  fun bookmarksTransacter(): Transacter
+
+  fun bookmarksQuery(limit: Long, offset: Long): Query<CatchUpItem>
+}
+
+internal class BookmarkRepositoryImpl(private val database: CatchUpDatabase) : BookmarkRepository {
+  private val scope = CoroutineScope(Dispatchers.IO)
```

Comment:
Inject this. Constructing this scope internally bakes in the assumption that the class is singleton-y; not doing this is a major reason we use the DI pattern and DI frameworks like Dagger.

Also, don't inject as a `val`. See comments below about launches on instance methods. Launches on initialization are fine IMO, and don't require making the scope a `val`, either.